### ISSUE TITLE
GPU Support (CUDA)

### ISSUE THREAD

**Issue by a-v-ebrahimi** [brandonmpetty/Doxa#27]
Do you have plan to support Nvidia CUDA?

**Comment by brandonmpetty** [comment]
I don't believe so.  My plan is to add support for the original version of AdOtsu and while doing so (75% there), do a slight refactor of the library to make it easier to use and more flexible for parameter analysis.  My final plan is to add Matlab bindings.  

At that point I will have exhausted almost every L.A.T. algorithm of note, and ported it to just about every system one might want to experiment with it in.  The last unexplored frontier would be CUDA/OpenCL, like you mentioned... or maybe SSE/AVX.

I am focusing right now on a large cyber security related project and because of that my time on this project has been cut short.  I am also not well versed in CUDA.  If you or anyone would like to contribute to this project, just let me know.  It may make more sense as a new repo, like Doxa-GPU, since it would be such a radical departure.  Something CPU based, like SSE/AVX, might make more sense living inside this specific repo.

**Comment by a-v-ebrahimi** [comment]
@brandonmpetty 
Thank for coming improvements!
My use case is binarizing old Arabic manuscripts.
No specific timing.
Yes, python integration is great.
